neo/rawio/plexonrawio.py

```diff
@@ -264,22 +267,49 @@ def _parse_header(self):
             signal_streams = np.array([], dtype=_signal_stream_dtype)
 
         else:
-            # detect groups (aka streams)
-            all_sig_length = all_sig_length = np.array(all_sig_length)
-            groups = set(zip(sig_channels["sampling_rate"], all_sig_length))
+            # Detect streams
+            all_sig_length = np.asarray(all_sig_length)
+
+            # names are WB{number}, FPX{number}, SPKCX{number}, AI{number}, etc
+            pattern = r"^\D+"  # Match any non-digit character at the beginning of the string
+            channels_prefixes = np.asarray([re.match(pattern, name).group(0) for name in sig_channels["name"]])
+            buffer_stream_groups = set(zip(channels_prefixes, sig_channels["sampling_rate"], all_sig_length))
+
+            # There are explanations of the streams based on channel names
+            # provided by a Plexon Engineer, see here:
+            # https://github.com/NeuralEnsemble/python-neo/pull/1495#issuecomment-2184256894
+            channel_prefix_to_stream_name = {
+                "WB": "WB-Wideband",
+                "FP": "FPl-Low Pass Filtered ",
+                "SP": "SPKC-High Pass Filtered",
+                "AI": "AI-Auxiliary Input",
+            }
+
+            # Using a mapping to ensure consistent order of stream_index
+            channel_prefix_to_stream_id = {
+                "WB": "0",
+                "FP": "1",
+                "SP": "2",
+                "AI": "3",
+            }
 
             signal_streams = []
             self._signal_length = {}
             self._sig_sampling_rate = {}
-            for stream_index, (sr, length) in enumerate(groups):
-                stream_id = str(stream_index)
+
+            for stream_index, (channel_prefix, sr, length) in enumerate(buffer_stream_groups):
+                # The users of plexon can modify the channel names, is not common but in that case
```

Comment:
```suggestion
                # The users of plexon can modify the prefix of the channel names (e.g. `my_prefix` instead of `WB`). This is not common but in that case
```

Reply:
@zm711

Reply:
Yep, that looks perfect!